PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/utils/AzureAbstractHandler.java

```diff
@@ -39,12 +39,19 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 
 		final Map<String, String> properties = new HashMap<>();
 		try {
-			properties.put("CategoryId", event.getCommand().getCategory().getId());
-			properties.put("Category", event.getCommand().getCategory().getName());
-			properties.put("CommandId", event.getCommand().getId());
-			properties.put("Text", event.getCommand().getName());
+			Command cmd = event.getCommand();
+			String handlerName = this.getClass().getSimpleName();
 			
-			AppInsightsClient.createByType(AppInsightsClient.EventType.Action, event.getCommand().getName(),	null, properties);
+			if (cmd != null) {
+				properties.put("CategoryId", cmd.getCategory().getId());
+				properties.put("Category", cmd.getCategory().getName());
+				properties.put("CommandId", cmd.getId());
+				properties.put("Text", cmd.getName());
+				if(null == handlerName || handlerName.isEmpty()) {
```

Comment:
`this` means the current object so it will NEVER be null. Same for its class name. I do think this check is unnecessary.